### PR TITLE
ref: move httpapi.Reponse into codersdk

### PR DESCRIPTION
**Pull request by sreya**
This PR just moves some logic around. It's bizarre for our sdk to be importing `httpapi` and `httpmw` both which should be viewed as packages internal to `coderd`. In general `codersdk` should host any logic that it needs to make requests or responses.

### CODE REVIEW

File: codersdk/client.go [hunk 1]
```diff
@@ -171,8 +171,8 @@ func readBodyAsError(res *http.Response) error {
 
 // Error represents an unaccepted or invalid request to the API.
 // @typescript-ignore Error
-type Error struct {
-	httpapi.Response
+type HTTPError struct {
```
Comment (kylecarbs) [inline codersdk/client.go:174]: I think this should still be `Error`. We don't prefix `Response` with HTTP, so this shouldn't either IMO.

**Review by kylecarbs — APPROVED** [review]
Good change. The import chaining was getting wacky.

File: codersdk/client.go (hunk 1 above)
Comment (sreya) [reply codersdk/client.go:174]: https://github.com/coder/coder/pull/2954/files#diff-7fad09d9d87a67ca91331d8399caa8b6644a66ae3b5fec5534cdeeff06a4b881R24

there's a conflict in naming when I moved this in, which is why i updated the name. I think it's appropriate given it contains http-specific fields

Comment (kylecarbs) [reply codersdk/client.go:174]: I see the diff now. I'd rather rename the other to `ValidationError` in that case. Seems more apt

**Pull request merged**